NetworkingFramework-main/TestFramework/HttpClient.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  HttpClient.swift
+//  TestFrameworkTests
+//
+//  Created by Дмитрий Игнатьев on 27.11.2021.
+//
+
+import UIKit
+
+
+//MARK: - Protocols-
+
+public protocol NetworkSession{
+    func dataTask(with url: URL, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> NetworkRequest
+}
+
+public protocol HTTPClient {
+    func get(from url: URL, completion: @escaping (Result<(Data, HTTPURLResponse), Error>) -> ())
+}
+
+public protocol NetworkRequest{
+    func resume()
+}
+
+extension URLSessionDataTask: NetworkRequest{
+}
+
+//MARK: - Classes-
+
+public class URLSessionHttpClient: HTTPClient{
+    private struct UnexpectedArguments: Error {}
+    private let session: NetworkSession
+    
+    public init(session: NetworkSession){
+        self.session = session
+    }
+    
+    public func get(from url: URL, completion: @escaping (Result<(Data, HTTPURLResponse), Error>) -> ()){
+        
+        let dataTask = session.dataTask(with: url){data, response, error in
```

Comment:
When we use trailing closure we one space between closure and the end of the method `( ` `{`

Reply:
done